lib/cc/analyzer/include_paths_builder.rb

```diff
@@ -16,153 +13,85 @@ def initialize(cc_exclude_paths, cc_include_paths = [])
         @cc_include_paths = cc_include_paths
       end
 
-      def build
-        if cc_include_paths.any?
-          paths = filter_by_cc_includes
+      def build(extra_excludes = [])
+        if extra_excludes.empty?
+          build_without_extra_excludes
         else
-          root = Directory.new(".", ignored_files)
-          paths = root.included_paths
-        end
-
-        paths.each do |path|
-          raise_on_unreadable_files(path)
+          build_with_extra_excludes(extra_excludes)
         end
       end
 
-      protected
-
-      def filter_by_cc_includes
-        paths = []
-        cc_include_paths.each do |path|
-          if File.directory?(path)
-            root = Directory.new(path, ignored_files)
-            paths += root.included_paths
-          elsif !ignored_files.include?(path)
-            paths << path
-          end
-        end
-        paths.uniq
-      end
+      private
 
-      def ignored_files
-        Tempfile.open(".cc_gitignore") do |tmp|
-          tmp.write(File.read(".gitignore")) if File.file?(".gitignore")
-          tmp << @cc_exclude_paths.join("\n")
-          tmp.close
-          tracked_and_ignored = `git ls-files -zi -X #{tmp.path} 2>/dev/null`.split("\0")
-          untracked_and_ignored = `git ls-files -zio -X #{tmp.path} 2>/dev/null`.split("\0")
-          tracked_and_ignored + untracked_and_ignored
-        end
+      def build_without_extra_excludes
+        @_build_without_extra_excludes ||= PathMinimizer.new(paths).minimize
```

Comment:
I think our house style is not to put leading underscores on ivars like this. i.e. just `@build_without_extra_excludes`